app/controllers/dishes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class DishesController < ApplicationController 
+  def show 
+    @dish = Dish.find(params[:id])
+  end
+
+  def update
+    ingredient = Ingredient.find(params[:ingredient_id])
+    dish = Dish.find(params[:id])
+    DishIngredient.create!({dish_id: dish.id, ingredient_id: ingredient.id})
+    
+    redirect_to "/dishes/#{dish.id}"
+  end
```

Comment:
The only thing I would call out is that this method, `update`, should be `create` and it should live on either the child controller `IngredientsController` or a join controller `DishIngredientsController`. If we think about what it's doing in the background it is creating a relationship between a child and the parent, it's not updating the parent.

The code itself is perfect.